### PR TITLE
refactor: initialize default config value by settings after plugin installation

### LINKED ISSUE #3160: 插件安装后初始化设置的默认值到 ConfigMap
### Your current Halo version

latest

### Describe this feature

如果插件添加了 Setting 资源，需要获取 Setting 定义的默认值到 ConfigMap，以避免用户需要点击保存才能使用的问题。

### Additional information

/kind improvement
/area core
/milestone 2.2.x
/assign

### PR DESCRIPTION
**Pull request by guqing**
#### What type of PR is this?
/kind improvement
/area core
/milestone 2.2.x

#### What this PR does / why we need it:
插件安装后根据配置的 settingName 读取默认值创建 ConfigMap
如果配置了 settingName 而没有配置 configMapName 则会自动填充为 uuid 并创建一个 ConfigMap
#### Which issue(s) this PR fixes:

Fixes #3160

#### Special notes for your reviewer:
/cc @halo-dev/sig-halo 
#### Does this PR introduce a user-facing change?

```release-note
插件安装后自动初始化 Setting 的默认值
```

### CODE REVIEW

File: src/main/java/run/halo/app/plugin/PluginCreatedEventReconciler.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            .switchIfEmpty(Mono.error(new IllegalStateException("Plugin not found: " + pluginName)))
+            .filter(plugin -> StringUtils.isNotBlank(plugin.getSpec().getSettingName()))
+            .filter(plugin -> StringUtils.isBlank(plugin.getSpec().getConfigMapName()))
+            .doOnNext(plugin -> {
+                // has settingName value but configMapName not configured
+                plugin.getSpec().setConfigMapName(UUID.randomUUID().toString());
+            })
+            .flatMap(client::update)
+            .doFinally(signalType -> latch.countDown())
+            .subscribe(service -> {
```
Comment (JohnNiang) [inline src/main/java/run/halo/app/plugin/PluginCreatedEventReconciler.java:91]: 其实在这里直接 block 就行了。我之前使用 CountDownLatch 的做法稍微有些复杂和繁琐。

**Review by JohnNiang — APPROVED** [review]
/approve

**Review by ruibaby — APPROVED** [review]
/lgtm

**Pull request merged**